File: LookinDemo/OC/LookinDemoOC/ViewController.m
//
//  ViewController.m
//  LookinDemoOC
//
//  Created by likai.123 on 2022/10/10.
//

#import "ViewController.h"

@interface TestView : UIImageView

@end

@implementation TestView

- (BOOL)lookin_shouldCaptureImage {
    return YES;
}

@end

@interface TestLayer : CALayer

@end

@implementation TestLayer

- (BOOL)lookin_shouldCaptureImage {
    return YES;
}

@end

@interface ViewController ()

@property(nonatomic, strong) TestView *myView;
@property(nonatomic, strong) TestLayer *myLayer;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.tag = 1111;
    // Do any additional setup after loading the view.
    self.myView = [TestView new];
    self.myView.image = [UIImage imageNamed:@"Logo_128"];
    self.myView.tag = 1234;
    [self.view addSubview:self.myView];
    self.myView.frame = CGRectMake(20, 20, 100, 100);
    
    self.myLayer = [TestLayer new];
    self.myLayer.backgroundColor = UIColor.blueColor.CGColor;
    [self.view.layer addSublayer:self.myLayer];
    self.myLayer.frame = CGRectMake(20, 200, 100, 100);
}

@end

@interface NSObject (LookinConfig)

@end

@implementation NSObject (LookinConfig)

+ (NSArray<NSString *> *)lookin_collapsedClassList {
    return @[@"UIDropShadowView"];
}

+ (NSDictionary<NSString *, UIColor *> *)lookin_colorAlias {
    return @{
        @"TitleColor": [UIColor colorWithRed:0.38 green:0.85 blue:0.22 alpha:1],
        @"MyWhite": [UIColor colorWithRed:1 green:1 blue:1 alpha:1],
        @"MyBlack": [UIColor colorWithRed:0 green:0 blue:0 alpha:1]
    };
}
+ (BOOL)lookin_shouldCaptureImageOfView:(UIView *)view {
    if (view.tag == 1234) {
        // Lookin will not show image of the view
        return NO;
    } else {
        return YES;
    }
}

+ (BOOL)lookin_shouldCaptureImageOfLayer:(CALayer *)layer {
//    if (...) {
//        // Lookin will not show image of the layer
//        return NO;
//    } else {
        return YES;
//    }
}

@end

File: LookinDemo/Swift/LookinDemoSwift/ViewController.swift
//
//  ViewController.swift
//  LookinDemoSwift
//
//  Created by likai.123 on 2022/10/10.
//

import UIKit

extension NSObject {
    @objc func lookin_shouldCaptureImageOfView(_ view: UIView) -> Bool {
        if view.tag == 1234 {
            return true
        } else {
            return true
        }
    }
    
    @objc func lookin_collapsedClassList() -> [String] {
        return ["UIDropShadowView"]
    }
    
    @objc func lookin_colorAlias() -> [String:UIColor] {
        return ["MyWhite": UIColor.init(red: 1, green: 1, blue: 1, alpha: 1)]
    }
}

class MyView: UIView {
    @objc func lookin_shouldCaptureImage() -> Bool {
        return false
    }
}

class ViewController: UIViewController {
    private let myButton = MyView()
    
    private let textView = UITextView()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.view.layer.borderColor = UIColor.red.cgColor
        
        textView.frame = CGRect(x: 20, y: 100, width: 200, height: 200)
        view.addSubview(textView)
        
        myButton.tag = 1234
        myButton.frame = CGRect(x: 20, y: 20, width: 50, height: 50)
        view.addSubview(myButton)
    }


}

